### PR TITLE
O3-2396: Appointments: Refine logic for whether a patient is checked …

### PR DESCRIPTION
**Pull request by mogoodrich**
…in or not

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

**Comment by mogoodrich** [comment]
see my comments here: https://openmrs.atlassian.net/browse/O3-2963 ... we can have broader discussions about making this change there.

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx [hunk 1]
```diff
@@ -22,13 +22,9 @@ interface AppointmentsActionsProps {
...
   const { t } = useTranslation();
   const { checkInButton, checkOutButton } = useConfig<ConfigObject>();
-  const { visits, mutateVisit } = useTodaysVisits();
+  const { mutateVisit } = useTodaysVisits();
   const patientUuid = appointment.patient.uuid;
   const visitDate = dayjs(appointment.startDateTime);
-  const hasActiveVisitToday = visits?.some((visit) => visit?.patient?.uuid === patientUuid && visit?.startDatetime);
-  const hasCheckedOutToday = visits?.some(
-    (visit) => visit?.patient?.uuid === patientUuid && visit?.startDatetime && visit?.stopDatetime,
-  );
```
Comment (mseaton) [inline packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:31]: As we discovered when looking at this today, this change can't be made in isolation of the changes to the checkout functionality, which currently will end the active visit.  Always ending the active visit makes sense when your checkout logic is based on the presence of an active visit.  But once you change the check-out logic to be independent of the visit status, you'll need to check that the visit is not already ended and/or prompt if the user wants to also close the visit, when you issue the modal dialog and perform the operation.

Comment (mogoodrich) [reply packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:31]: Yeah, I likely want to tweak this.

**Comment by mogoodrich** [comment]
Converting this to a draft, we don't want to merge this in until we properly are able to switch appointment actions via the default paths

**Pull request closed without merging**